packages/ckeditor5-modern/src/ModernEditor/ModernEditor.tsx

```diff
@@ -59,10 +66,68 @@ const ModernEditor: FC<ModernEditorProps> = ({
 
   const { isSubmitting } = formState;
   const values = getValues();
+
   const images = useMemo(() => files.map(file => URL.createObjectURL(file)), [
     files,
   ]);
 
+  const uploadFiles = async (files: File[]) => {
+    try {
+      setUploadLoading(true);
+      setUploadCurrentLoading(0);
+      setUploadTotalLoading(files.length * 100);
+      let currentTotal: number[] = [];
+
+      // get upload credentials
+      const resultUploadCredentials: Promise<string>[] = files.map(
+        async (file: File) => {
+          return await uploadGetCrendentials({
+            type: file.type.split('/')[0],
+            filename: file.name,
+            apiUrl: uploadConfig?.apiUrl || '',
+            file: file,
+          });
+        }
+      );
+
+      const dataUploadCredentials = await Promise.all(resultUploadCredentials);
+      const url = dataUploadCredentials.map((upload: any) => upload.data.url);
+
+      // upload the file with credentials
+      const resultUpload = dataUploadCredentials.map(
+        async (upload: any, index: number) => {
+          const { data } = upload;
+          const formData = new FormData();
+
+          Object.keys(data.params).forEach(key => {
+            formData.append(key, data.params[key]);
+          });
+          formData.append('Content-Type', upload.file.type);
+          formData.append('file', upload.file);
+
+          return await uploadFile({
+            apiUrl: data.origin || '',
+            data: formData,
+            onLoadProgress: (total: number) => {
+              currentTotal[index] = total;
+              setUploadCurrentLoading(currentTotal.reduce((a, b) => a + b, 0));
+            },
+          });
+        }
+      );
```

Comment:
@nberongoy avoid using `any` and omit `async` & `await` here just like on the previous request.